### ISSUE TITLE
VNC not working

### ISSUE THREAD

**Issue by peter12908** [Saleh7/ProxmoxVE_PHP_API#8]
Code:
```
$node = 'AoW';
$vmid = 501;
$f = Nodes::createQemuVncproxy($node, $vmid, $data = array());
$array = json_decode(json_encode($f), True);
Nodes::qemuVncwebsocket($node, $vmid, $array["data"]["port"], $array["data"]["ticket"]);
$vnc = 'https://xxx.xxx.xxx.xxx:8006/?console=kvm&novnc=1&node='.$node.'&resize=scale&vmid='.$vmid.'&path=api2/json/nodes/'.$node.'/qemu/'.$vmid.'/vncwebsocket?port=' . $array["data"]["port"] . '&vncticket=' . $array["data"]["ticket"];
echo $vnc;
echo '<iframe src="'.$vnc.'" width="1280" height="720" allowfullscreen="true"></iframe>';
```
it always says: Failed to connect to server

**Comment by emresaracoglu** [comment]
I don't think proxmox works compatible with this script anymore with new API codes.

**Comment by peter12908** [comment]
i am using Proxmox VE 5.4 and everything works just fine except this vnc thing

**Comment by rmdhfz** [comment]
same issue

**Comment by AnushK-Fro** [comment]
`    public function vnc() {
        $acl = [
            'username' => 'User username',
            'password' => 'User password',
            'realm' => 'pve'
        ];

        $ticket = Access::createTicket($acl);
        $ticket = json_encode($ticket, true);
        $ticket = json_decode($ticket, true);

        setcookie('PVEAuthCookie', $ticket['data']['ticket'], 0, '/', 'PROXMOXDOMAIN', false);

        return print("<iframe width=\"100%\" height=\"100%\" frameborder=\"0\" allowfullscreen=\"\" src=\"https://PROXMOXHOSTNAME:8006/?console=kvm&novnc=1&node=proxmox&resize=scale&vmid=$server->vm_id\"></iframe>");
    }`

Note with this code, the Proxmox panel and Laravel panel must be under the same domain. Also, please create an isolated user for each user. If the user goes to VNC, they can access the Proxmox panel too.

**Comment by AVSISP** [comment]
6 Years later and we still don't have a solution for a secure VNC loading without giving the user full proxmox panel access? This is just insane...

**Comment by AnushK-Fro** [comment]
> 6 Years later and we still don't have a solution for a secure VNC loading without giving the user full proxmox panel access? This is just insane...

It’s possible with a reverse proxy. See this repository for example:

https://github.com/ConvoyPanel/coterm

You can also setup a Proxmox user and only set its permissions so that it can only access VNC on 1 particular virtual machine.

**Comment by AVSISP** [comment]
> > 6 Years later and we still don't have a solution for a secure VNC loading without giving the user full proxmox panel access? This is just insane...
> 
> It’s possible with a reverse proxy. See this repository for example:
> 
> https://github.com/ConvoyPanel/coterm
> 
> You can also setup a Proxmox user and only set its permissions so that it can only access VNC on 1 particular virtual machine.

Yeah, that is highly ingrained into assumptions of Convoy, which I wouldn't be using this PHP module if I was using. I think the same applies to most others.

I ended up just making my own proxy in python and running it as a service on the website's server - being sure to include token auth and so on + encoding the node name + vmid into an encrypted, time-limited token and everything like that. Took  while - but really shouldn't have to be this difficult.

**Comment by AnushK-Fro** [comment]
> Yeah, that is highly ingrained into assumptions of Convoy, which I wouldn't be using this PHP module if I was using. I think the same applies to most others.

Yes, it was just an example. It's open source so you could take a peek at the code.
